iam/__tests__/snapshotProposalsProvider.test.ts

```diff
@@ -0,0 +1,177 @@
+// ---- Test subject
+import { RequestPayload } from "@gitcoin/passport-types";
+import { SnapshotProposalsProvider, snapshotGraphQLDatabase } from "../src/providers/snapshotProposalsProvider";
+
+// ----- Libs
+import axios from "axios";
+
+jest.mock("axios");
+const mockedAxios = axios as jest.Mocked<typeof axios>;
+
+const MOCK_ADDRESS = "0xcF314CE817E25b4F784bC1f24c9A79A525fEC50f";
+const MOCK_ADDRESS_LOWER = MOCK_ADDRESS.toLocaleLowerCase();
+const NO_MOCK_ADDRESS = "";
+const BAD_MOCK_ADDRESS = "F314CE817E25b4F784bC1f24c9A79A525fEC50f";
+const BAD_MOCK_ADDRESS_LOWER = BAD_MOCK_ADDRESS.toLocaleLowerCase();
+
+const validSnapshotResponse = {
+  data: {
+    data: {
+      proposals: [
+        {
+          id: "0x8f46d1bd6d14681d42fa01b80f14b56f2953f5a8b95154d6ce9c8fe1db599771",
+          scores_total: 0,
+          author: `${MOCK_ADDRESS_LOWER}`,
+        },
+        {
+          id: "0x4d33fc4d9f6b8a6ed44ef78a1cc3dcb6c120c7f28fa938ced99385bbdaa0ba23",
+          scores_total: 228,
+          author: `${MOCK_ADDRESS_LOWER}`,
+        },
+      ],
+    },
+  },
+};
+
+const invalidSnapshotResponseNoProposalVotes = {
+  data: {
+    data: {
+      proposals: [
+        {
+          id: "0x8f46d1bd6d14681d42fa01b80f14b56f2953f5a8b95154d6ce9c8fe1db599771",
+          scores_total: 0,
+          author: `${MOCK_ADDRESS_LOWER}`,
+        },
+        {
+          id: "0x4d33fc4d9f6b8a6ed44ef78a1cc3dcb6c120c7f28fa938ced99385bbdaa0ba23",
+          scores_total: 0,
+          author: `${MOCK_ADDRESS_LOWER}`,
+        },
+      ],
+    },
+  },
+};
+
+const invalidSnapshotResponseNoProposals = {
+  data: {
+    data: {
+      proposals: null as [],
+    },
+  },
+};
+
+interface RequestData {
+  query: string;
+}
+
+describe("Attempt verification", function () {
```

Comment:
I would also add a test like:

`it("should return invalid payload an exception is thrown when request is made",`